### DISCUSSION TITLE
Inquiry about CLDNN and MKL-DNN Dependencies in the Latest OpenVINO Version

### DISCUSSION THREAD

**Discussion by max-my** [openvinotoolkit/openvino#26473] [Q&A] [answered]
I am reaching out to inquire about the dependencies within the latest version of OpenVINO. Our team is in the process of assessing OpenVINO for deployment in our AI inference workflows and has specific interests in the underlying libraries that contribute to the optimization process.

1. Does the current release of OpenVINO depend on the Compute Library for Deep Neural Networks (CLDNN) and the Math Kernel Library for Deep Neural Networks (MKL-DNN)?
2. If these libraries are still part of the stack, could you please outline their role and how they are integrated within the OpenVINO framework?
Understanding the dependency structure and the utilization of these libraries is essential for us to ensure that our deployment aligns with the performance and compatibility standards we require.

Thank you for your time and assistance. We are excited about the potential of leveraging OpenVINO in our projects and are looking forward to engaging with the community.

**Comment by libohao666** [comment]
Hello, I'm a colleague of max-my and also an AI and deep learning enthusiast. I came across your inquiry regarding the dependencies on CLDNN and MKL-DNN in the latest version of OpenVINO, and I believe it's a very pertinent and timely question. Understanding the current integration of these libraries is crucial for our team as we evaluate and deploy OpenVINO.

Our team is also considering integrating OpenVINO into our projects, so we're very interested in the role and performance impact of these underlying optimization libraries.

I'm looking forward to more discussions and insights from the community members and developers on this topic. It would be incredibly helpful for us to hear from those with hands-on experience.

Thanks for bringing this up, and I appreciate the OpenVINO community for providing this platform for exchange.

**Comment by ilya-lavrenov** [comment, marked as answer]
clDNN is not available as library / dependency anymore. Its code now part of plugin, common plugin infra. Kernels from former clDNN are now used for integrated GPUs, while oneDNN for GPU is used to provide kernels for discrete GPUs.

oneDNN for CPU is used in CPU plugin as provider of optimized CPU kernels. In OpenVINO we use our own fork of oneDNN https://github.com/openvinotoolkit/onednn, which has multiple extra optimization of top of upstream version of oneDNN. Ultimately, we want to get rid of this fork and merge these 2 versions and this work is ongoing. 

If you want to contribute to underlying optimizations, I would recommend to create a ticket / feature request and our team can help to analyze which places need to be optimized and suggest you a plan what kernels / parts you can work on.

**Reply by max-my** [reply]
Thank you for the clarification on the integration of clDNN into the plugin infrastructure and the use of oneDNN for optimized kernels. It's great to see the progress being made in the project.

I appreciate the work that's being done to merge the oneDNN forks, and I'm looking forward to seeing the benefits this will bring to the community.
